scripts/json-app-tool.php

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env php
+<?php
+
+// Pulled from includes/polling/functions.inc.php
+function data_flatten($array, $prefix = '', $joiner = '_')
+{
+    $return = array();
+    foreach ($array as $key => $value) {
+        if (is_array($value)) {
+            if (strcmp($prefix, '')) {
+                $key=$prefix.$joiner.$key;
+            }
+            $return = array_merge($return, data_flatten($value, $key, $joiner));
+        } else {
+            if (strcmp($prefix, '')) {
+                $key=$prefix.$joiner.$key;
+            }
+            $return[$key] = $value;
+        }
+    }
+    return $return;
+}
+
+// Pulled from includes/snmp.inc.php
+function string_to_oid($string)
+{
+    $oid = strlen($string);
+    for ($i = 0; $i != strlen($string); $i++) {
+        $oid .= '.'.ord($string[$i]);
+    }
+
+    return $oid;
+}//end string_to_oid()
+
+
+// Options!
+$short_opts='sktmlhj:a:';
+$options = getopt($short_opts);
+
+// print the help
+if (isset($options[h])) {
+    echo "LibreNMS JSON App tool
+  -j      The file containing the JSON to use for the test.
+  -s      Print the SNMPrec data.
+  -t      Print the JSON test data file.
+  -l      Just load and lint the JSON file.
+  -m      Extract and print metric variables from the JSON file.
+  -k      If m is specified, just print the keys in tested order.
+  -a      The application name for use with -s and -t.
+  -h      Show this help text.
+
+-j must always be specified.
+
+-a must always be given with -s and -t.
+
+-l if specified will override any others. If none of the others are
+   specified and just -j is given, then the file is loaded and then
+   linted if needed. For linting jsonlint needs to be installed.
+
+-m is handy if you want to grab a list of metrics a JSON app is returning
+   and how they data is flattned as well for when writing alert rules. -k
+   prints it in a slightly neater manner and in a manner and in tested
+   order.
+";
+    exit();
+}
+
+// make sure we have a JSON file to work with
+if (!isset($options[j])) {
```

Comment:
This needs to be a string, not a constant. `$options[j]` -> `$options['j']`.  Same as everywhere.  This will cause code to fail in PHP 7.3 fyi ;)

Reply:
Ohh! Thanks dude! That is great to know. :) Fixed.